### PR TITLE
fix(dx): eslint all files

### PR DESCRIPTION
**Pull request by blimmer**
Today, linting fails in `server` and `client`:

```
> pwd
parabol/packages/server
> yarn lint
yarn run v1.22.18
$ eslint --fix "**/*.{ts,tsx}"

/Users/blimmer/code/parabol/parabol/packages/server/graphql/types/UserTiersCount.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: graphql/types/UserTiersCount.ts.
The file must be included in at least one of the projects provided

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
```

```
> pwd
parabol/packages/client
> yarn lint
yarn run v1.22.18
$ eslint --fix "**/*.{ts,tsx}"

/Users/blimmer/code/parabol/parabol-2/packages/client/components/AtmosphereProvider/PortalProvider.tsx
  6:57  error  Unexpected empty arrow function  @typescript-eslint/no-empty-function

/Users/blimmer/code/parabol/parabol-2/packages/client/components/Dashboard/DashNavTeam.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: components/Dashboard/DashNavTeam.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/components/FacilitatorMenu.tsx
  22:77  error  Expected '!==' and instead saw '!='  eqeqeq

/Users/blimmer/code/parabol/parabol-2/packages/client/components/JiraFieldMenu.tsx
  29:28  warning  React Hook "useMemo" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks

/Users/blimmer/code/parabol/parabol-2/packages/client/components/MeetingCard.tsx
  172:27  warning  React Hook "useBreakpoint" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?           react-hooks/rules-of-hooks
  173:60  warning  React Hook "useMenu" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?                 react-hooks/rules-of-hooks
  174:15  warning  React Hook "useAnimatedMeetingCard" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks
  186:7   warning  React Hook "useTooltip" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?              react-hooks/rules-of-hooks

/Users/blimmer/code/parabol/parabol-2/packages/client/components/MeetingHelp/ActionMeetingLobbyHelpMenu.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: components/MeetingHelp/ActionMeetingLobbyHelpMenu.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/components/MeetingHelp/RetroLobbyHelpMenu.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: components/MeetingHelp/RetroLobbyHelpMenu.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/components/RetroReflectPhase/PhaseItemEditor.tsx
  89:73  error  Expected '===' and instead saw '=='  eqeqeq

/Users/blimmer/code/parabol/parabol-2/packages/client/components/ScopePhaseArea.tsx
  94:37  warning  React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks

/Users/blimmer/code/parabol/parabol-2/packages/client/components/TaskEditor/Ellipsis.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: components/TaskEditor/Ellipsis.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/components/TaskEditor/useCommentPlugins.ts
  66:25  error  Unexpected empty method 'useTaskChild'  @typescript-eslint/no-empty-function

/Users/blimmer/code/parabol/parabol-2/packages/client/components/promptResponse/PromptResponseEditor.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: components/promptResponse/PromptResponseEditor.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/hooks/useAllIntegrations.ts
  40:23  error  Expected '===' and instead saw '=='  eqeqeq
  42:30  error  Expected '===' and instead saw '=='  eqeqeq

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/Button.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/Button.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/EmailBlock/EmailBlock.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/EmailBlock/EmailBlock.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/EmailFooter/EmailFooter.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/EmailFooter/EmailFooter.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/EmailVerificationEmail.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/EmailVerificationEmail.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/EmptySpace/EmptySpace.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/EmptySpace/EmptySpace.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/Header/Header.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/Header/Header.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/Layout/Layout.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/Layout/Layout.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/MeetingSummaryEmailRootSSR.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/MeetingSummaryEmailRootSSR.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/NotificationSummaryEmail.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/NotificationSummaryEmail.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/ResetPasswordEmail.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/ResetPasswordEmail.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/EmailExportCSVButton.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/SummaryEmail/MeetingSummaryEmail/EmailExportCSVButton.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/MeetingMembersWithoutTasks.tsx
  49:16  warning  React Hook "useEmailItemGrid" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/RetroTopics.tsx
  90:24  warning  React Hook "useEmailItemGrid" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/SummaryEmail/SummaryCTA.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/SummaryEmail/SummaryCTA.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/SummaryEmail/SummaryEmailScheduleCalendar.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/SummaryEmail/SummaryEmailScheduleCalendar.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/TeamInvite.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/TeamInvite.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/email/components/UpcomingInvoiceEmail.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/email/components/UpcomingInvoiceEmail.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/meeting/components/ActionMeetingUpdatesPromptViewerHelpText.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/meeting/components/ActionMeetingUpdatesPromptViewerHelpText.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/meeting/components/MockScopingTask.tsx
  43:47  error  Unexpected empty arrow function  @typescript-eslint/no-empty-function

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/teamDashboard/components/LeaveTeamModal/LeaveTeamModal.tsx
  25:66  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/teamDashboard/components/ProviderRow/AtlassianProviderRow.tsx
  42:85  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/teamDashboard/components/ProviderRow/GitHubProviderRow.tsx
  38:85  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/teamDashboard/components/TeamSettings/TeamSettings.tsx
  39:66  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/teamDashboard/containers/UnpaidTeamModal/UnpaidTeamModalRoot.tsx
  19:66  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/userDashboard/components/LeaveOrgModal/LeaveOrgModal.tsx
  19:85  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/userDashboard/components/OrgBilling/OrgBillingUpgrade.tsx
  41:43  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/userDashboard/components/Organization/Organization.tsx
  61:45  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/userDashboard/components/RemoveFromOrgModal/RemoveFromOrgModal.tsx
  19:85  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/blimmer/code/parabol/parabol-2/packages/client/modules/userDashboard/components/UserDashMain/DebugButton.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: modules/userDashboard/components/UserDashMain/DebugButton.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/mutations/fragments/CompleteOrganizationFrag.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: mutations/fragments/CompleteOrganizationFrag.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/mutations/fragments/CompleteTaskFrag.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: mutations/fragments/CompleteTaskFrag.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/mutations/fragments/UserAnalyticsFrag.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: mutations/fragments/UserAnalyticsFrag.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/polyfills.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: polyfills.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/AtlassianIntegrationId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/AtlassianIntegrationId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/EstimateUserScoreId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/EstimateUserScoreId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/GQLExecutorChannelId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/GQLExecutorChannelId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/GitHubIntegrationId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/GitHubIntegrationId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/IntegrationHash.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/IntegrationHash.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/IntegrationProviderId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/IntegrationProviderId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/IntegrationRepoId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/IntegrationRepoId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/JiraDimensionFieldId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/JiraDimensionFieldId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/JiraServerIssueId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/JiraServerIssueId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/MeetingMemberId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/MeetingMemberId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/PollId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/PollId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/PollOptionId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/PollOptionId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/SocketServerChannelId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/SocketServerChannelId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/TeamMemberId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/TeamMemberId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/TeamMemberIntegrationAuthId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/TeamMemberIntegrationAuthId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/TeamMemberIntegrationsId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/TeamMemberIntegrationsId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/shared/gqlIds/TemplateDimensionRefId.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: shared/gqlIds/TemplateDimensionRefId.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/stories/components/StoryProvider.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: stories/components/StoryProvider.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/stories/helpers/Incrementor.tsx
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: stories/helpers/Incrementor.tsx.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/date/formatWeekday.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/date/formatWeekday.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/draftjs/getTypeFromEntityMap.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/draftjs/getTypeFromEntityMap.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/draftjs/removeEntityKeepText.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/draftjs/removeEntityKeepText.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/githubQueries/createIssue.graphql.d.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/githubQueries/createIssue.graphql.d.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/githubQueries/getProfile.graphql.d.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/githubQueries/getProfile.graphql.d.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/githubQueries/getRepoInfo.graphql.d.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/githubQueries/getRepoInfo.graphql.d.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/githubQueries/getRepos.graphql.d.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/githubQueries/getRepos.graphql.d.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/meetings/computeNewScaleValue.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/meetings/computeNewScaleValue.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/screenBugs/Bug.ts
  77:13  error  Expected '===' and instead saw '=='  eqeqeq

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/smartGroup/getGroupMatrix.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/smartGroup/getGroupMatrix.ts.
The file must be included in at least one of the projects provided

/Users/blimmer/code/parabol/parabol-2/packages/client/utils/smartGroup/groupReflections.ts
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: utils/smartGroup/groupReflections.ts.
The file must be included in at least one of the projects provided

✖ 80 problems (72 errors, 8 warnings)
```

This PR updates the eslint tsconfig file to parse all files. It also adds a CI step to run linting to identify lint problems at build-time. Since I added the CI step, I also fixed all outstanding errors.

### CODE REVIEW

File: packages/client/components/AtmosphereProvider/PortalProvider.tsx [hunk 1]
```diff
@@ -3,7 +3,9 @@ import {createPortal} from 'react-dom'
 import useForceUpdate from '../../hooks/useForceUpdate'
 
 export type SetPortal = (id: string, portal: ReactNode) => void
-export const PortalContext = React.createContext((() => {}) as SetPortal)
+export const PortalContext = React.createContext((() => {
+  return
```
Comment (blimmer) [inline packages/client/components/AtmosphereProvider/PortalProvider.tsx:7]: Currently, [`no-empty-function`](https://typescript-eslint.io/rules/no-empty-function/) is enforced. So, to retain the existing behavior and make this rule happy, I `return`.

File: packages/client/components/FacilitatorMenu.tsx [hunk 2]
```diff
@@ -19,25 +19,35 @@ const FacilitatorMenu = (props: Props) => {
   const {id: meetingId, facilitatorUserId, meetingMembers} = meeting
   const atmosphere = useAtmosphere()
   const {viewerId} = atmosphere
-  const facilitatorCandidateIds = meetingMembers.filter(({user}) => user.id != facilitatorUserId && user.isConnected).map(({user}) => user.id)
+  const facilitatorCandidateIds = meetingMembers
+    .filter(({user}) => user.id !== facilitatorUserId && user.isConnected)
```
Comment (blimmer) [inline packages/client/components/FacilitatorMenu.tsx:23]: This is the only change. `==` becomes `===`.

File: packages/client/tsconfig.eslint.json [hunk 3]
```diff
@@ -1,14 +1,5 @@
 {
   "extends": "./tsconfig.json",
```
Comment (blimmer) [inline packages/client/tsconfig.eslint.json:2]: I don't have context on why a separate `tsconfig.eslint.json` is used here and in `server`. Typically, you just have a single `tsconfig.json` that's used throughout.

File: packages/client/components/AtmosphereProvider/PortalProvider.tsx (hunk 1 above)
Comment (Dschoordsch) [reply packages/client/components/AtmosphereProvider/PortalProvider.tsx:7]: +1 I prefer to put a comment `// noop` there to emphasize it's intentional

**Review by Dschoordsch — APPROVED** [review]
I still get a couple of warnings
```
parabol-client: ✖ 8 problems (0 errors, 8 warnings)
```

**Comment by mattkrick** [comment]
doing a quick spotcheck, i think these might be fixed already because running `yarn lint` gets me some different errors.
rather than do this in CI where the results won't be persisted, i'd like to do this in a github action.
see #6314 #6315

**Pull request closed without merging**